staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go

```diff
@@ -210,6 +229,40 @@ func (a *Webhook) ShouldCallHook(h webhook.WebhookAccessor, attr admission.Attri
 	return invocation, nil
 }
 
+func (a *Webhook) evaluateCELRules(h webhook.WebhookAccessor, attr admission.Attributes, o admission.ObjectInterfaces) ([]cel.PolicyDecision, error) {
+	//TODO ivelichkovich: How expensive is compile? This is a very high traffic path. ValidatingAdmissionPolicy has a controller run a loop refreshing and compiling
+	validator := a.celValidatorCompiler.Compile(
+		h.GetFailurePolicy(),
+		nil,
+		matchConditionsToValidations(h.GetMatchConditions()),
+	)
+	versionedAttr, err := NewVersionedAttributes(attr, attr.GetKind(), o)
+	if err != nil {
+		//TODO ivelichkovich: if error do we just continue? This currently forces no call to webhook
+		klog.Errorf("ivelichkovich failed to convert object version: %w", err)
+		return nil, nil
+	}
+	decisions, err := validator.Validate(versionedAttr, nil)
+	if err != nil {
+		//TODO ivelichkovich: if error do we just continue? This currently moves to calling webhook
+		klog.Error("ivelichkovich error in webhook cel validation")
+		return nil, nil
+	}
+	return decisions, nil
+}
+
+// TODO ivelichkovich: should I use some common object here rather than v1alpha.Validation?
```

Comment:
Yes. IMO the following should be done:

- `CELValidator` -> `CELEvaluator`
- `Validate` -> `Evaluate`. Refactor the method to extract the PolicyDecision logic, and just return a raw list of `evalResults`, or maybe a custom type that adds a bit more context to the results
- Maybe still have a `CELValidator` that wraps a `CELEvaluator`, and encapsulates the PolicyDecision logic

Would be good to get input from @jpbetz and others working on CEL Admission first though.